bin/arrayNormalization.R

```diff
@@ -267,24 +267,21 @@ illuminaArray <- function(files, outFile) {
 
 	print("Finished reading in Illumina files")
 
-
-	print("Background correcting")
-
-	# The 'bgAdjust' method will estimate the background based on the 
-	# control probe information.
-	lumi.B <- lumiB(lumiBatch, method='bgAdjust')
-
-
 	# Many data coming to ArrayExpress and GEO having missing the STDEV or STDERR columns 
 	# (depends on you BeadStudio version), which are required by LumiBatch class. 
 	# When this data is missed these columns, the lumiR function will output the 
 	# ExpressionSet object instead of LumiBatch object. And you can't do VST transform because it. 
 	# A safe parameters to go ahead with is log2 transformation.
-	if ( any(assayDataElementNames(lumi.B) == "se.exprs") ) {
-		lumi.T <- lumiT(lumi.B, method = 'vst')
-		} else {
-			lumi.T <- lumiT(lumi.B, method = 'log2')
-	}  
+	# Function lumiT performs variance stabilizing transform (VST) with both input and output being LumiBatch object.
+	# VST - Variance stabilization is critical for subsequent statistical inference to identify differential genes from microarray data
+ 	if ( any(assayDataElementNames(lumiBatch) == "se.exprs") ) {
```

Comment:
Is it better to have this heuristic than to to just do log2 with everything? I'm inclined to think the consistency is more important than getting the analysis slightly better in a subset of cases. What do you think @pcm32 ?

Reply:
-  Many data coming to ArrayExpress and GEO having missing the STDEV or STDERR columns  (depends on you BeadStudio version), which are required by LumiBatch class. It is assumed that they are background corrected.
- When this data is missed these columns, the lumiR function will output the ExpressionSet object instead of LumiBatch object. Then, a safe parameters to go ahead with is log2 transformation.
- However,  when STDERR fields is provided then performs background correction and VST transformation.